tests/level-db/test_leveldb_db_backend.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from evm.db.backends.memory import MemoryDB
+from evm.db import (
+    get_db_backend,
+)
+
+
+pytest.importorskip('leveldb')
+
+
+# Sets db backend to leveldb
+@pytest.fixture
+def config_env(monkeypatch):
+    monkeypatch.setenv('CHAIN_DB_BACKEND_CLASS',
+                       'evm.db.backends.level.LevelDB')
+
+
+@pytest.fixture(scope='session')
```

Comment:
The reason your test is failing is a combination from using `LevelDB.db` and scoping this fixture as `"session"`.

So, by using `LevelDB.db`, you're tests are all using the same database instance (meaning that any writes you make to the database will persist between test runs).  

Then, since you are using a *session* level fixture, pytest doesn't remove the database between tests (which it should since you want to test runs to be completely isolated.

So, to fix this, make the suggested changes to the `LevelDB` class code and remove the scoping of this fixture.

I suspect that the reason it fails on travis and not locally is that the tests are not being run in the same order on your local machine and on travis.